lib/commands/new.js

```diff
@@ -29,6 +30,27 @@ module.exports = Command.extend({
     '<app-name>'
   ],
 
+  beforeRun: function(rawArgs) {
+    if (!rawArgs.length) {
+      return;
+    }
+    // merge in blueprint availableOptions
+    var blueprint;
+    try{
+      blueprint = this.lookupBlueprint(rawArgs[0]);
+      this.registerOptions(blueprint);
+    }
+    catch(e) {
+      SilentError.debugOrThrow('ember-cli/commands/init', e);
```

Comment:
Yes, agreed. I'm new to ember-cli development, and am still figuring out the patterns. Creating a base class with the method seems like overkill. A new module in utilities perhaps?

```javascript
// utilities/merge-blueprint-options.js
function mergeBlueprintOptions(command, rawArgs) {
  ...
}
```

and

```javascript
import mergeBlueprintOptions from '../utilities/merge-blueprint-options';

module.exports = Command.extend({
  ...
  beforeRun: function(rawArgs) {
    mergeBlueprintOptions(this, rawArgs);
  }
  ...
});
```